test/cpp/api/tensor.cpp

```diff
@@ -194,6 +196,80 @@ TEST(TensorTest, ContainsCorrectValuesForManyValuesVariable) {
   ASSERT_TRUE(almost_equal(tensor[2], 3.125));
 }
 
+TEST(TensorTest, ContainsCorrectValuesForMultidimValuesVariable) {
+  {
+    auto tensor = torch::tensor({{1, 2}, {3, 4}});
+    ASSERT_EQ(tensor.dtype(), torch::kInt);
+    ASSERT_EQ(tensor.sizes(), torch::IntArrayRef({2, 2}));
+    ASSERT_TRUE(torch::allclose(tensor, torch::arange(1, 5, torch::kInt).view(tensor.sizes())));
+    ASSERT_FALSE(tensor.requires_grad());
```

Comment:
nit: might be nice to write a helper function that does something like:
- assert dtypes equal
- assert sizes equal
- assert tensors allclose

The closest thing in python we have is `self.assertEqual`, which checks all of the above except for 'assert dtypes equal' (I think that is a bug, but that's another story).

Reply:
I feel that writing the asserts explicitly might help with having finer control of what checks are being run. Using a helper function here would make it simpler, but we would need to worry about what the helper function checks every time we try to use it.

Reply:
In most of these cases the code is asking "is this tensor equivalent to this other tensor that I have" and a helper function would be good to catch that. But yes if a lot of the checks will be special cased depending on the helper functions then it's better to have individual asserts so the logic on what is being checked is clearer